runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -129,44 +133,35 @@ public synchronized void acquireBuffer() {
         if (serializedCache == null) {
             serializedCache = Unpooled.buffer();
             doSerializeInternal(serializedCache);
+            lastKnownSize = serializedCache.readableBytes();
         } else {
             serializedCache.retain();
         }
     }
 
     @Override
     public int getSizeEstimate() {
-        byte[] tempData = data;
-        if (tempData != null) {
-            return tempData.length;
-        }
-
         if (lastKnownSize != NOT_KNOWN) {
             return lastKnownSize;
         }
-
         return 1;
     }
 
     @Getter
-    final EnumMap<LogUnitMetadataType, Object> metadataMap;
+    private final ByteBuf serializedMetadataMap = Unpooled.buffer();
 
```

Comment:
I suppose we still need metadataMap that is null at the server side and only get populated at the client/runtime side.

Reply:
I did some benchmark with the serialized version. It did save some space. With a 8 streams transaction: 
LogData in the cache (metaDataMap is null):getSizeEstimate 543 deepSize 1568
Same LogData at Runtime (metaDataMap is not null: getSizeEstimate 543 deepSize 4584
As now the LogData at the Cache is a flat data structure, we can still sizeOf.deepSize to get its accurate size. 

Please add a test case and also do a benchmark to see if there are performance issues. You may refer my branch for the testcase : xq/logUnitCache01

Reply:
ok, will do.  Right now I am hitting errors in existing unit tests

Reply:
I fixed some issues in my branch xq/logUnitCache01. Replace serializedMetadata with byte[]. The problem with ByteBuf is that after eaching reading, the readerIndex will be changed, the further read operations will hit the out of boundary exception. Another issue is that if there are two threads are reading the bytebuf at the same time, it messes the readerIndex.

Reply:
Thanks!  Yes I replaced it similarly in an earlier patch.